test/integration/gateway_test.go

```diff
@@ -382,6 +382,8 @@ func TestUnmanagedGatewayControllerSupport(t *testing.T) {
 		unsupportedGateway, err = gatewayClient.GatewayV1beta1().Gateways(ns.Name).Get(ctx, unsupportedGateway.Name, metav1.GetOptions{})
 		require.NoError(t, err)
 		require.Len(t, unsupportedGateway.Status.Conditions, 1)
+		//lint:ignore SA1019 the default upstream reason is still NotReconciled https://github.com/kubernetes-sigs/gateway-api/pull/1701
+		//nolint:staticcheck
```

Comment:
Can you explain what is the reason for the `nolint` and `lint` directives here?

Reply:
It's an upstream problem: the `NotReconciled` reason is marked as deprecated, but it's still used as default in the CRD. If you go to the linked PR, you can see it. Here the linters are complaining because we are using a deprecated Reason.